sdk/nodejs/runtime/closure/codePaths.ts

```diff
@@ -18,6 +18,52 @@ import * as readPackageTree from "read-package-tree";
 import * as asset from "../../asset";
 import { RunError } from "../../errors";
 
+/**
+ * computeRequiredSubDependencyPaths Takes in the set of "require"d modules from a serialied function
+ * and determines which dependencies of @pulumi/... need to be included.  Normally We do not include
+ * @pulumi/... packages in a serialized function (largely due to the size as well as because these
+ * libraries are only intended to be used at deployment time.  However, @pulumi/... might itself
+ * depend on *other* libraries that *are* needed at runtime.  This function helps determine that set
+ * so that we will include those sub-packages even if we filter out the main @pulumi/... package
+ * itself.
+ */
+export function computeRequiredSubDependencyPaths(requiredModules: Set<string>): Set<string> {
```

Comment:
This doesn't quite feel like the right composable primitive to be introducing.  It's odd that we have to tie this in some way to `@pulumi/...`, it seems there is a more general problem here where code that gets "moved" from a library into the root will not be able to directly see node_modules that are nested inside the original library's `node_modules`.  

We should discuss whether there is a more fundamental change needed to support this well, and how to make this notion of leaving out "deployment time-only packages" more robust.